src/frontend/contexts/auth-context.js

```diff
@@ -1,23 +1,49 @@
 import { createContext, useContext, useState } from "react";
 import { useNavigate } from "react-router-dom";
 import axios from "axios";
+import { toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
 
 const AuthContext = createContext();
 export const useAuth = () => useContext(AuthContext);
 
 export function AuthProvider({ children }) {
   const navigate = useNavigate();
-  const initialState = localStorage.getItem("token");
-  const [token, setToken] = useState(initialState || "");
+  const [token, setToken] = useState(localStorage.getItem("token") || "");
   const [userData, setUserData] = useState({});
-  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [isLoggedIn, setIsLoggedIn] = useState(
+    localStorage.getItem("isLoggedIn") || false
+  );
   const [loginInput, setLoginInput] = useState({});
 
+  const notifyLogin = () =>
+    toast.success("Login successful", {
+      position: "top-right",
+      autoClose: 2000,
+      hideProgressBar: false,
+      closeOnClick: true,
+      pauseOnHover: true,
+      draggable: true,
+      progress: undefined,
+    });
+
+  const notifyLogout = () =>
+    toast.success("Logout successful", {
+      position: "top-right",
+      autoClose: 2000,
+      hideProgressBar: false,
+      closeOnClick: true,
+      pauseOnHover: true,
+      draggable: true,
+      progress: undefined,
+    });
```

Comment:
Suggestion: You can make one utility function and just pass the msg you want in toast, as all properties are small except msg.